pulsar-metadata/src/main/java/org/apache/pulsar/metadata/bookkeeper/PulsarLedgerAuditorManager.java

```diff
@@ -47,12 +49,20 @@ class PulsarLedgerAuditorManager implements LedgerAuditorManager {
         this.leaderElection =
                 coordinationService.getLeaderElection(String.class, electionPath, this::handleStateChanges);
         this.leaderElectionState = LeaderElectionState.NoLeader;
+        store.registerSessionListener(event -> {
+            if (SessionEvent.SessionLost == event) {
+                sessionExpired = true;
+                synchronized (lockObj) {
+                    notifyAll();
+                }
+            }
+        });
```

Comment:
As mentioned in the PR's description

```
There is a mechanism in place: when a SessionLost event occurs, AutoRecovery initiates a shutdown, which closes the Auditor.
```

If the AutoRecovery will initiate a shutdown operation. Will it call PulsarLedgerAuditorManager close() method? If yes, I think we'd better handle this case in the close() method instead of handle the session lost event for both AutoRecovery and PulsarLedgerAuditorManager.

Reply:
> If the AutoRecovery will initiate a shutdown operation. Will it call PulsarLedgerAuditorManager close() method? If yes, I think we'd better handle this case in the close() method instead of handle the session lost event for both AutoRecovery and PulsarLedgerAuditorManager.

At now, it will invoke the close() method, but the shutdown behavior executes in the `AuditorElector` executor. 
The problem is that the `AuditorElector` executor is already blocked due to the election. 
So the shutdown task can't be executed.

https://github.com/apache/bookkeeper/blob/50e30db96ab411135202d5da56e2f7e171d32193/bookkeeper-server/src/main/java/org/apache/bookkeeper/replication/AuditorElector.java#L150C2-L168C6